deepcell_tracking/isbi_utils.py

```diff
@@ -40,33 +40,48 @@
 from deepcell_tracking.utils import load_trks
 
 
-def trk_to_isbi(track, path):
+def trk_to_isbi(track, path=None):
     """Convert a lineage track into an ISBI formatted text file.
 
     Args:
         track (dict): Cell lineage object.
-        path (str): Path to save the .txt file.
-    """
-    with open(path, 'w') as text_file:
-        for label in track:
-            first_frame = min(track[label]['frames'])
-            last_frame = max(track[label]['frames'])
-            parent = track[label]['parent']
-            parent = 0 if parent is None else parent
-            if parent:
-                parent_frames = track[parent]['frames']
-                if parent_frames[-1] != first_frame - 1:
-                    parent = 0
+        path (str): Path to save the .txt file (deprecated).
 
+    Returns:
+        array: Array of ISBI dictionaries of each label.
+    """
+    if path is not None:
+        print("UserWarning: The `path` argument is deprecated.")
+        text_file = open(path, 'w')
```

Comment:
To ensure the file closes properly every time, its best to use a `with` context manager. and do all the file-writing inside of it. To make it easier/to use, let's just conditionally open/write the file after you've already built your list. (In another for loop over the `isbi` list).